File: src/main/java/com/company/project/service/TodoService.java
package com.company.project.service;

import java.util.List;

import com.company.project.beans.Items;

public interface TodoService {

	List<Items> getAllItems();

	Items findById(int id);

	boolean isItemExists(Items item);

	void saveItem(Items item);

	void updateItem(Items currentItem);

	void deleteItem(int id);

	void deleteAllItems();

}

File: src/main/resources/static/js/app/TodoService.js
'use strict'
angular.module('demo.services', []).factory('TodoService',
		[ "$http", "CONSTANTS", function($http, CONSTANTS) {
			var service = {};
			service.getItemById = function(id) {
				var url = CONSTANTS.getItemByIdUrl + id;
				return $http.get(url);
			}
			service.getAllItems = function() {
				return $http.get(CONSTANTS.getAllItems);
			}
			service.saveItem = function(item) {
				return $http.post(CONSTANTS.saveItem, item);
			}
			service.deleteItem = function(id) {
				return $http.get(CONSTANTS.deleteItem + id);
			}
			service.updateItem = function(item) {
				return $http.post(CONSTANTS.updateItem, item);
			}
			return service;
		} ]);